discovery/store.go

```diff
@@ -271,6 +272,12 @@ func (s *sqlStore) search(serviceID string, query map[string]string, allowUnvali
 	if !allowUnvalidated {
 		stmt = stmt.Where("validated != 0")
 	}
+	// remove wildcards to prevent unneeded join on credential
+	for k, v := range query {
```

Comment:
this modified the given query, which is error-prone. Better build a new query map